keyboards/helix/rev3_4rows/keymaps/default/keymap.c

```diff
@@ -0,0 +1,311 @@
+   * |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |             |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * |      | Reset|RGBRST|EEPRST|      |      |             |      |      |      |      |      |  Del |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * |      |      |      |      |      | Mac  |             | Win  |      |RGB ON| HUE+ | SAT+ | VAL+ |
+   * |------+------+------+------+------+------+------+------+------+------+------+------+------+------|
+   * |      |      |      |      |      |      |      |      |      |      | MODE | HUE- | SAT- | VAL- |
+   * `-------------------------------------------------------------------------------------------------'
+   */
+  [_ADJUST] =  LAYOUT( \
+      KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,                     KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12, \
+      _______, RESET,   RGBRST,  EEP_RST, _______, _______,                   _______, _______, _______, _______, _______, KC_DEL, \
+      _______, _______, _______, _______, _______, AG_NORM,                   AG_SWAP, _______, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, \
+      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD \
+      )
+
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* Left side encoder */
+        if (clockwise) {
+            tap_code(KC_PGDN);
+        } else {
+            tap_code(KC_PGUP);
+        }
+    } else if (index == 1) { /* Right side encoder */
+        if (clockwise) {
+            tap_code(KC_DOWN);
+        } else {
+            tap_code(KC_UP);
+        }
+    }
+}
+
+void dip_switch_update_user(uint8_t index, bool active) {
+    switch (index) {
+        case 0:
+            if(active) { // Left no.1
+              keymap_config.swap_lalt_lgui = true;
+            } else {
+              keymap_config.swap_lalt_lgui = false;
+            }
+            break;
+        case 1:
+            if(active) { // Left no.2
+              ;
+            } else { }
+            break;
+        case 2:
+            if(active) { // Right no.2
+              ;
+            } else { }
+            break;
+        case 3:
+            if (active) { // Right no.1
+              ;
+            } else { }
+            break;
+    }
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case EISU:
+      if (record->event.pressed) {
+        if(keymap_config.swap_lalt_lgui==false){
+          register_code(KC_LANG2);
+        }else{
+          SEND_STRING(SS_LALT("`"));
```

Comment:
For short macros, tap_code is better, as the compiled size for it is smaller than SEND_STRING. 
```suggestion
          tap_code16(LALT(KC_GRAVE));
```